Example-iOS-Swift/WebViewController.swift

```diff
@@ -38,16 +77,46 @@ class WebViewController: UIViewController, WKScriptMessageHandler {
         webView.allowsBackForwardNavigationGestures = true
     }
     
+    struct MessageData: Decodable {
+        let source: String?
+        let eventName: String?
+        let data: Dictionary<String, String>?
+    }
+    
     func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
-        avatarUrlDelegate?.avatarUrlCallback(url : "\(message.body)")
-        reloadPage(clearHistory: false)
+        
+        if let body = message.body as? String{
+            
+            let jsonData = body.data(using: .utf8)
+            
+            if let bodyStruct = try? JSONDecoder().decode(MessageData.self, from: jsonData!){
+                switch bodyStruct.eventName {
+                case "v1.avatar.exported":
+                    avatarUrlDelegate?.avatarUrlCallback(url : "\(String(describing: bodyStruct.data?["url"]))")
+                    reloadPage(clearHistory: false)
+                case "v1.subscription.created":
+                    subscriptionCreated = true
+                default:
+                    print("Default event. bodyStruct = \(bodyStruct)")
+                }
+            }
+            else if (!subscriptionCreated) {
+                if (body.contains("glb")){
```

Comment:
Same here but less concerning since you check for V1 in the end. Still, maybe it's safer to check for endsWith?

Reply:
Yep good idea, ill change to the swift equivalent 
`body.hasSuffix("glb")`

Reply:
updated